eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -312,9 +312,7 @@ private[channel] object ChannelCodecs3 {
     /** Once a dual funding tx has been signed, we must remember the associated commitments. */
     case class DualFundingTx(fundingTx: SignedSharedTransaction, commitments: Commitments)
 
-    private val dualFundingTxCodec: Codec[DualFundingTx] = (
-      ("fundingTx" | signedSharedTransactionCodec) ::
-        ("commitments" | commitmentsCodec)).as[DualFundingTx]
+    private val dualFundingTxCodec: Codec[DualFundingTx] = fail[DualFundingTx](Err("you have been running dual funding before it was officially released! contact developers"))
```

Comment:
I think you don't even need to bother with this error or backwards-compat' here, nobody should be using dual-funding in the wild yet since eclair and cln aren't even compatible at that point.

Reply:
Why? Can't you just modify the `dualFundingTxCodec` without caring about backwards-compat?

Reply:
Oh I see we can't because we've previously been clever with the `dualFundingTxCodec` and made it work with the previous codec for funding txs